src/game/gui/textinput.c

```diff
@@ -212,6 +213,7 @@ static void textinput_free(component *c) {
     textinput *tb = widget_get_obj(c);
     surface_free(&tb->sur);
     str_free(&tb->text);
+    text_objects_free(tb->text_cache, (sizeof(tb->text_cache) / sizeof(tb->text_cache[0])));
```

Comment:
Why isn't the second parameter an integer?

Reply:
size_t is an unsigned int, but will be 64bit or 32bit based on your architecture. And it's not a signed int because I never expect to have a negative amount of objects to destroy.

Reply:
I mean why not just pass the number of text objects

Reply:
size_t is fine, but the arraysize magic should probably be replaced with a constant (same as the [0]'s, [1]'s, [2]'s and [3]'s, as pointed out in a different conversation)

Reply:
This should become a macro called ARRAY_SIZE(tb->text_cache).
That makes it forward compatible, when someone expands or reduces the array, they won't need to change this instance.